client/src/main/java/io/pravega/client/stream/notifications/notifier/EndOfDataEventNotifier.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.stream.notifications.notifier;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Supplier;
+
+import io.pravega.client.state.StateSynchronizer;
+import io.pravega.client.stream.impl.ReaderGroupState;
+import io.pravega.client.stream.notifications.Listener;
+import io.pravega.client.stream.notifications.NotificationSystem;
+import io.pravega.client.stream.notifications.events.EndOfDataEvent;
+import javax.annotation.concurrent.GuardedBy;
+import lombok.Synchronized;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class EndOfDataEventNotifier extends AbstractEventNotifier<EndOfDataEvent> {
+    private static final int UPDATE_INTERVAL_SECONDS = Integer.parseInt(
+            System.getProperty("pravega.client.endOfDataEvent.poll.interval.seconds", String.valueOf(120)));
+    private final Supplier<StateSynchronizer<ReaderGroupState>> synchronizerSupplier;
+    private final AtomicBoolean pollingStarted = new AtomicBoolean();
+    @GuardedBy("$lock")
+    private ScheduledFuture<?> future;
+    @GuardedBy("$lock")
+    private StateSynchronizer<ReaderGroupState> synchronizer;
+
+    public EndOfDataEventNotifier(final NotificationSystem notifySystem,
+                                  final Supplier<StateSynchronizer<ReaderGroupState>> synchronizerSupplier,
+                                  final ScheduledExecutorService executor) {
+        super(notifySystem, executor);
+        this.synchronizerSupplier = synchronizerSupplier;
+    }
+
+    @Override
+    @Synchronized
+    public void registerListener(final Listener<EndOfDataEvent> listener) {
```

Comment:
These register/unregister methods are identical to the ones for segment event, we should consider duplicating the code and moving it to an abstract class.

Reply:
Refactored the code.